// This file was generated with the Protobuf generator tool.
// Dataset ID    : 40693
// Dataset Name  : xd6;
// Dataset URL   : https://www.openml.org/data/v1/download/4965270/xd6.arff
// Num. Columns  : 10
// Num. Rows     : 973
// Target Feature: class

// Beginning of Description of Dataset:
// **Author**: Unknown  
// **Source**: [PMLB](https://github.com/EpistasisLab/penn-ml-benchmarks/tree/master/datasets/classification) - Supposedly originates from UCI, but can't find it there anymore.  
// **Please cite:**  
// **XD6 Dataset**
// Dataset used by Buntine and Niblett (1992). Composed of 10 features, one of which is irrelevant. The target is a disjunctive normal form formula over the nine other attributes, with additional classification noise.
// [More info](https://books.google.be/books?id=W2bmBwAAQBAJ&pg=PA313&lpg=PA313&dq=dataset+xd6&source=bl&ots=6hYPdz8_Nl&sig=TR1ieOg9D1pCrvNyeKbb-3eKmd8&hl=en&sa=X&ved=0ahUKEwj_tZ_MxozZAhVHa1AKHZVEBBsQ6AEIQjAF#v=onepage&q=dataset xd6&f=false).
//

syntax = "proto3";

message Empty {
}

message Features {
	string   Attribute_1                    = 1;
	string   Attribute_2                    = 2;
	string   Attribute_3                    = 3;
	string   Attribute_4                    = 4;
	string   Attribute_5                    = 5;
	string   Attribute_6                    = 6;
	string   Attribute_7                    = 7;
	string   Attribute_8                    = 8;
	string   Attribute_9                    = 9;
	string   Class                          = 10;
}

service get_next_row {
	 rpc get_next_row(Empty) returns(Features);
}

/*
//This code section can be directly used
//at the end of gen_next_row in the server.py file
//for OpenML dataset nr. 40693
        response.Attribute_1                    = row[0]
        response.Attribute_2                    = row[1]
        response.Attribute_3                    = row[2]
        response.Attribute_4                    = row[3]
        response.Attribute_5                    = row[4]
        response.Attribute_6                    = row[5]
        response.Attribute_7                    = row[6]
        response.Attribute_8                    = row[7]
        response.Attribute_9                    = row[8]
        response.Class                          = row[9]
*/

